src/content/posts/2023-03-09-creating-a-pagination-component-with-astro.md

```diff
@@ -0,0 +1,288 @@
+
+Something that Hugo did well was Pagination. It was very simple to implement but it came with some limitations. Astro does not have a specific Pagination component but it _does_ give you the tools to make your own. Here's what I came up with and used on a couple of our sites for blog sections...
+
+## [TLDR;](#putting-it-together)
+<div class="mb-3 text--center">
+<img src="/images/get-to-the-point.gif" alt="Get to the point"/>
+</div>
+
+First, create your blog page within `/src/pages`. I like to add another folder here called `/blog` just to try to keep things a bit cleaner. So we have:
+
+```
+src
+- pages
+  - blog
+```
+
+Inside here add your page for the blog landing. This is the page that will show all of your posts, or at least what we want. You'll need the use brackets in this page title. Call it `[...page].astro`. We need the `[]` wrapping the name as this is a dynamic route.
+
+```
+src
+- pages
+  - blog
+    - [...page].astro
+```
+
+## [Blog Landing](#blog-landing)
+Now let's get to the guts of `[...page].astro`. You'll be leveraging `page.data` here. We also need 2 other components `PostSummary.astro`, which is a block of blog info like a title, image, and a teaser. You can add more to this if you need and I'll show you what this looks like in a bit. We also need a `Pagination.astro` component. This can also be customizzed to your needs but it will most likely always have the same data passed to it.
+
+The frontmatter of `[...page].astro` should look like:
+
+```javascript
+---
+import PostSummary from "../../layouts/partials/blog/PostSummary.astro";
+import Pagination from '../../components/Pagination.astro';
+
+export async function getStaticPaths({ paginate }) {
+  const allPosts = await Astro.glob('./*.md*');
+  const formattedPosts = allPosts.sort((a, b) => new Date(a.frontmatter.date) - new Date(b.frontmatter.date)).reverse();
+
+  return paginate(formattedPosts, {
+    pageSize: 5,
+  })
+}
+const { page } = Astro.props;
+---
+```
+
+Include your components, PostSummary and Pagination. We use the `getStaticPaths` function to grab each blog post. In this case it's anything in our /blog folder and we're looking for any file that ends in either a `.md` or `.mdx` file type. We want to format them by the most recent post. That's what we're doing with the `formattedPosts` variable. Once we have our posts array we can set the total number of posts we want to show. In this case it's 5. 
+
+```javascript
+  return paginate(formattedPosts, {
+    pageSize: 5,
+  })
+```
+
+Finally we need to get page data so we use: 
+
+```javascript
+const { page } = Astro.props;
+```
+
+If you want to see what this looks like add `console.log(page)` in your front matter. This will show you an array of your posts and the data inside them you can utilize.
```

Comment:
```suggestion
If you want to see what this looks like, add `console.log(page)` in your front matter. This will show you an array of your posts and the data inside them that you can utilize.
```